hardware/Makefile.machine.include

```diff
@@ -0,0 +1,47 @@
+# Copyright (c) 2019, University of Washington All rights reserved.
+# 
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+# 
+# Redistributions of source code must retain the above copyright notice, this list
+# of conditions and the following disclaimer.
+# 
+# Redistributions in binary form must reproduce the above copyright notice, this
+# list of conditions and the following disclaimer in the documentation and/or
+# other materials provided with the distribution.
+# 
+# Neither the name of the copyright holder nor the names of its contributors may
+# be used to endorse or promote products derived from this software without
+# specific prior written permission.
+# 
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+# ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# HARDWARE_PATH: The path to the hardware folder in BSG F1
+ifndef HARDWARE_PATH
+$(error $(shell echo -e "$(RED)BSG MAKE ERROR: HARDWARE_PATH is not defined$(NC)"))
+endif
+
+KNOWN_MACHINE_TYPES = $(notdir $(shell find $(HARDWARE_PATH) -maxdepth 1 -mindepth 1 -type d))
+# MACHINE_TYPE: The type of machine to target.
+ifndef MACHINE_TYPE
+$(error $(shell echo -e "$(RED)BSG MAKE ERROR: MACHINE_TYPE is not defined$(NC)"))
+endif
+
+# Confirm that MACHINE_TYPE is valid
+ifeq ($(filter $(MACHINE_TYPE),$(KNOWN_MACHINE_TYPES)),)
+$(info $(shell echo -e "$(RED)BSG MAKE ERROR: Unknown MACHINE_TYPE $(MACHINE_TYPE)"))
+$(error $(shell echo -e "$(RED)BSG MAKE ERROR: Valid MACHINE_TYPE values are $(KNOWN_MACHINE_TYPES)"))
+endif
+
+HARDWARE_MACHINE_PATH = $(HARDWARE_PATH)/$(MACHINE_TYPE)
```

Comment:
Using MACHINE_TYPE we include the appropriate machine.mk fragment from the machine directory.

Reply:
The downside of using MACHINE_TYPE is that it needs to be defined in a file somewhere. I haven't quite figured out where the best place to define it is.

Reply:
Define a default machine type here and leave it for user to override it on command-line?

Reply:
Yes, this would be a solution. 

The problem is that MACHINE_TYPE can differ between F1 and cosimulation. Is it better to define MACHINE_TYPE for each? Or is it better to define MACHINE_TYPE in one place?

That's the essential conflict of my question